app/screens/integration_selector/index.tsx

```diff
@@ -0,0 +1,609 @@
+        delete multiselectSelectedItems[itemKey];
+        setMultiselectSelected(multiselectSelectedItems);
+    };
+
+    const getChannels = useCallback(debounce(async () => {
+        if (next.current && !loading && !term) {
+            setLoading(true);
+            page.current += 1;
+
+            const {channels: channelData} = await fetchChannels(serverUrl, currentTeamId, page.current);
+
+            setLoading(false);
+
+            if (channelData && channelData.length > 0) {
+                setIntegrationData(channelData);
+            } else {
+                next.current = false;
+            }
+        }
+    }, 100), [integrationData, loading, term]);
+
+    const getProfiles = useCallback(debounce(async () => {
+        if (next.current && !loading && !term) {
+            setLoading(true);
+            page.current += 1;
+
+            const {users: profiles} = await fetchProfiles(serverUrl, page.current);
+
+            setLoading(false);
+
+            if (profiles && profiles.length > 0) {
+                setIntegrationData(profiles);
+            } else {
+                next.current = false;
+            }
+        }
+    }, 100), [integrationData, loading, term]);
+
+    const loadMore = useCallback(async () => {
+        if (dataSource === ViewConstants.DATA_SOURCE_USERS) {
+            await getProfiles();
+        } else if (dataSource === ViewConstants.DATA_SOURCE_CHANNELS) {
+            await getChannels();
+        }
+
+        // dynamic options are not paged so are not reloaded on scroll
+    }, [getProfiles, getChannels, dataSource]);
+
+    const searchDynamicOptions = useCallback(async (searchTerm = '') => {
+        if (options) {
+            setIntegrationData(options);
+        }
+
+        if (!getDynamicOptions) {
+            return;
+        }
+
+        const results: DialogOption[] = await getDynamicOptions(searchTerm.toLowerCase());
+        const searchData = results || [];
+
+        if (searchTerm) {
+            setSearchResults(searchData);
+        } else {
+            setIntegrationData(searchData);
+        }
+    }, [options, getDynamicOptions]);
+
+    const onHandleMultiselectSubmit = useCallback(() => {
+        handleSelect(getMultiselectData());
+        close();
+    }, [multiselectSelected, handleSelect]);
+
+    const onSearch = useCallback((text: string) => {
+        if (!text) {
+            clearSearch();
+        }
+
+        setTerm(text);
+
+        if (searchTimeoutId.current) {
+            clearTimeout(searchTimeoutId.current);
+        }
+
+        searchTimeoutId.current = setTimeout(async () => {
+            if (!dataSource) {
+                setSearchResults(filterSearchData('', integrationData, text));
+                return;
+            }
+
+            setLoading(true);
+
+            if (dataSource === ViewConstants.DATA_SOURCE_USERS) {
+                const {data: userData} = await searchProfiles(
+                    serverUrl, term.toLowerCase(),
```

Comment:
If you are setting the `term` to the`text` on line 295, shouldn't you continue using the `text` here (and other references to term in this function)? If so, you probably can remove the term from the dependencies.